src/components/elements/text/text.tsx

```diff
@@ -52,14 +79,26 @@ export const Text = (props: IGenericElementProps<ITextSchema>) => {
 	};
 
 	return (
-		<Element
-			id={id}
-			data-testid={getTestId(id)}
-			{...otherSchema}
-			// NOTE: Parent text body should be transformed into <div> to prevent validateDOMNesting error
-			{...(hasNestedFields() && { as: "div" })}
-		>
-			<Sanitize id={id}>{renderText()}</Sanitize>
-		</Element>
+		<>
+			<Element
+				id={id}
+				ref={contentDivRef}
+				maxLines={maxLines > 0 && !isMore ? maxLines : undefined}
+				data-testid={getTestId(id)}
+				{...otherSchema}
+				// NOTE: Parent text body should be transformed into <div> to prevent validateDOMNesting error
+				{...(hasNestedFields() && { as: "div" })}
+			>
+				<Sanitize id={id}>{renderText()}</Sanitize>
+			</Element>
+
+			<div>
+				{!hideButton && maxLines > 0 && (
```

Comment:
why need to check maxLines > 0?
shouldn't `hideButton` be true if maxLines is falsey?

Reply:
noted, removed